Error message when fetching pipeline id fails

sk1p opened the pull request:
I was getting [another silent failure](https://github.com/LiberTEM/LiberTEM/actions/runs/17095073254/job/48477486719); the last output was "Triggered CI for branch refs/heads/master", so I think it must have been a failure to fetch the pipeline id.

It would be nice of course to capture the actual error message of the last `curl` run instead, but I didn't find an easy way to implement that in a bash script.

sk1p committed:
Error message when fetching pipeline id fails

jakob-fritz commented:
Thank you for that proposal. I didn't see at first glance, why the pipeline-id could not be fetched. However, it should be reported, as you said.

If you want to, you can rerun that job and check if the issue persists. If so, we can also look into it together, if you want to.

jakob-fritz commented on get_ci_state.sh:
```diff
@@ -36,6 +36,7 @@ do
   pipeline_id_attempt=$((pipeline_id_attempt + 1))
   if [ "$pipeline_id_attempt" -gt "5" ]
   then
+    echo "Failed to fetch pipeline id"
```
I would suggest to also include the used SHA in the output. This allows to double-check if the issue arises from a wrong/strangely chosen SHA. It furthermore allows for easier checking in the remote repo, as the commit-SHA in question does not need to be searched for.

```suggestion
    echo "Failed to fetch pipeline id for ${used_sha}"
```

jakob-fritz reviewed (approved):
I propose to explicitly add the used SHA to the error message. This allows for easier hunting of the underlying issue.

jakob-fritz committed:
Add SHA of mirror to output

This output is shown before a the script exits with a failure. Therefore, the output shall make finding the reason easier and the failure more explicit.

sk1p commented:
Thanks the review and merging! SHA in the output sounds good. 

> Thank you for that proposal. I didn't see at first glance, why the pipeline-id could not be fetched. However, it should be reported, as you said.
> 
> If you want to, you can rerun that job and check if the issue persists. If so, we can also look into it together, if you want to.

The issue didn't persist, I think it was due to intermittent network failure between github and the gitlab instance.
